File: Algorithm/여러가지문제/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> days(progresses.size());
    queue<int> q;
    for(int i = 0; i < progresses.size(); i++){
        while( !(progresses[i] >= 100)){   
           progresses[i] += speeds[i];
           days[i]++;
       }
    }
    for(int i = 0; i < days.size(); i++){
        q.push(days[i]);
    }
    
    int cnt = 1;
    int curr = q.front();
    q.pop();
    while (!q.empty()) {
        if (q.front() <= curr) {
            q.pop();
            cnt++;
        }
        else {
            answer.push_back(cnt);
            cnt = 1;
            curr = q.front();
            q.pop();
        }
    }
    answer.push_back(cnt);

    return answer;
}

File: Algorithm/여러가지문제/기능개발.py

def solution(progresses, speeds):
    answer = []
    depDay = []
    
    for task, speed in zip(progresses, speeds):
        d, m = divmod(100 - task, speed)
        if m != 0:
            d += 1
        depDay.append(d)
    front = 0
    for idx in range(len(depDay)):
        if depDay[front] < depDay[idx]:
            answer.append(idx-front)
            front = idx
    answer.append(len(depDay) - front)
    return answer